go/consensus/cometbft/light/service.go

```diff
@@ -254,14 +256,70 @@ func (c *client) GetLightBlock(ctx context.Context, height int64) (*consensus.Li
 		return nil, nil, ctx.Err()
 	}
 
-	// Try local backend first.
-	lb, err := c.consensus.GetLightBlock(ctx, height)
-	if err == nil {
-		return lb, rpc.NewNopPeerFeedback(), nil
+	// Local backend source.
+	localBackendSource := func() (*consensus.LightBlock, rpc.PeerFeedback, error) {
+		lb, err := c.consensus.GetLightBlock(ctx, height)
+		if err == nil {
+			return lb, rpc.NewNopPeerFeedback(), nil
+		}
+
+		c.logger.Debug("failed to fetch light block from local full node",
+			"err", err,
+			"height", height,
+		)
+
+		return nil, nil, err
+	}
+
+	// Light client.
+	lightClientSource := func() (*consensus.LightBlock, rpc.PeerFeedback, error) {
+		clb, err := c.lc.GetVerifiedLightBlock(ctx, height)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		plb, err := clb.ToProto()
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to marshal light block: %w", err)
+		}
+		meta, err := plb.Marshal()
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to marshal light block: %w", err)
+		}
+
+		return &consensus.LightBlock{
+			Height: clb.Height,
+			Meta:   meta,
+		}, rpc.NewNopPeerFeedback(), nil
+	}
+
+	// Direct peer query.
+	directPeerQuerySource := func() (*consensus.LightBlock, rpc.PeerFeedback, error) {
+		return c.lc.GetLightBlock(ctx, height)
+	}
+
+	// Try all sources in order.
+	var mergedErr error
+	for srcIdx, src := range []func() (*consensus.LightBlock, rpc.PeerFeedback, error){
+		localBackendSource,
+		lightClientSource,
+		directPeerQuerySource,
+	} {
+		lb, pf, err := src()
+		if err == nil {
+			return lb, pf, nil
+		}
+
+		c.logger.Debug("failed to fetch light block from source",
+			"err", err,
+			"source", srcIdx,
```

Comment:
```suggestion
			"source_idx", srcIdx,
```

Reply:
I would remove this and try to log this in the sources, using the name instead of the idx. In couple of months, I will forgot the order and will have to look at the source code to find out which source failed.